subprojects/language-java/src/main/java/org/gradle/api/tasks/compile/JavaCompile.java

```diff
@@ -282,19 +282,31 @@ private JavaToolchainSpec determineExplicitToolchain() {
     }
 
     /**
-     * The Groovy source-classes mapping file. Internal use only.
+     * The source-classes mapping file. Internal use only.
      *
      * @since 6.5
      */
-    @LocalState
+    @OutputFile
     protected File getSourceClassesMappingFile() {
         if (sourceClassesMappingFile == null) {
-            File tmpDir = getServices().get(TemporaryFileProvider.class).newTemporaryFile(getName());
-            sourceClassesMappingFile = new File(tmpDir, "source-classes-mapping.txt");
+            sourceClassesMappingFile = new File(getTemporaryDir(), "source-classes-mapping.txt");
```

Comment:
:o: I wonder if we should move the file _out_ of the temporary directory, now that it's a real output.

Reply:
I think it's fine like this as long as we consider this an internal detail of the task. If we ever make this configurable or let users read our analysis file (effectively making it API), then I'd agree putting it in a more prominent, configurable location.

Reply:
Note that the "temp" dir is still in the build folder, not in `/tmp`. So it's not temporary at all :)